src/persist/src/indexed/mod.rs

```diff
@@ -41,7 +41,68 @@ use crate::indexed::encoding::{
 };
 use crate::indexed::metrics::Metrics;
 use crate::pfuture::PFutureHandle;
-use crate::storage::{Blob, Log, SeqNo};
+use crate::storage::{Blob, BlobRead, Log, SeqNo};
+
+/// WIP
+#[derive(Debug)]
+pub enum CmdRead {
+    /// WIP
+    Load(String, (String, String), PFutureHandle<Option<Id>>),
```

Comment:
I vaguely recall some uneasy about how Register works but I've forgotten the details (you wanted it to take in both the string and id maybe?). now that we have this, I wonder if we could change the create_or_load protocol to fix that

Reply:
I think it was a pretty abstract concern.

basically

if l had a sequence of commands like this in a log

```
load ("foo" ...)
load ("bar" ...)
write(Id(X), <data>)
write(Id(Y), <data2>)
```

and I told you "apply this sequence of commands to an empty persist runtime, and then tell me the states of all the collections" you would need some way to know which `Id(_)` mapped to which external stream name / would have to have access to the internals of how `persist` manages that. It's a tiny bit unsatisfying because the rest of the commands are all "self-describing" enough that anyone could replay them back and know exactly what to do.

I don't think this is a practical concern now or in the forseeable future.

Reply:
ah right. well I think this helps move in that direction! we could replace Register with something that takes both the name and Id, make Load return the next free Id if it doesn't find it, and the impl create_or_load using a combination of those two (with care to handle races of course)